### PR TITLE
🎉(all) initial commit

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose
Setup Hugo and blowfish. Add dev, QA tools.

overview

## Proposal

- [x] Install hugo
- [x] Install a theme (blowfish)
- [x] Setup & Customize theme to Open FUN
- [x] Add dev tools
- [x] Add QA tools
- [x] Add github pages deployment job

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: Build
+        run: hugo --minify
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        if: github.ref == 'refs/heads/main'
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./public
```
Comment (quitterie-lcs) [inline .github/workflows/gh-pages.yml:31]: An endline is missing

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+/assets/jsconfig.json
+hugo_stats.json
+
+# -- Executable may be added to repository
+hugo.exe
+hugo.darwin
+hugo.linux
+
+# -- Temporary lock file while building
+/.hugo_build.lock
```
Comment (quitterie-lcs) [inline .gitignore:13]: Ditto

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+down: ## stop and remove container
+	@$(COMPOSE) down
+.PHONY: down
+
+logs: ## display hugo logs
+	@$(COMPOSE) logs -f hugo
+.PHONY: logs
+
+status:
```
Comment (quitterie-lcs) [inline Makefile:38]:
```suggestion
status: ## an alias for "docker-compose ps"
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+# -- Articles
+new-post: ## Create a new post
+	bin/new-post
+.PHONY: new-post
+
+# -- Misc
+help:
+	@grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+.PHONY: help
```
Comment (quitterie-lcs) [inline Makefile:80]: Ditto

File: README.md [hunk 5]
```diff
@@ -1,5 +1,76 @@
...
+make new-post
+```
+
+Then fill the prompt by providing the post title :
+```sh
+What is the title of the new post?
+>>>
+```
+
+Once the post created, you will find the fresh new post into
```
Comment (quitterie-lcs) [inline README.md:39]:
```suggestion
Once the post created, post content is stored into
```

File: README.md [hunk 6]
```diff
@@ -1,5 +1,76 @@
...
+````
+Next time you can simply use the command:
+```sh
+make run
+```
+to start the Hugo development server.
+
+Once this is done, Hugo is served on http://localhost:1313 🎉.
+
+You are now able to write content! Take a look at the [next section](#authoring) to know how to do.
```
Comment (quitterie-lcs) [inline README.md:23]:
```suggestion
You are now able to write content! Have a look at the [next section](#authoring) to know how to do.
```

File: data/authors/jbpenrath.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  "name": "Jean-Baptiste PENRATH",
+  "image" : "authors/jbpenrath.jpg",
+  "bio": "Web Developer at France Université Numérique",
+  "social": [
+    { "linkedin": "https://www.linkedin.com/in/jean-baptiste-penrath-73473a62" },
+    { "twitter": "https://twitter.com/Hi4s" },
+    { "github": "https://github.com/jbpenrath" },
+    { "codepen": "https://codepen.io/jbpenrath/" }
+  ]
+}
```
Comment (quitterie-lcs) [inline data/authors/jbpenrath.json:11]: Ditto

File: data/authors/template.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    { "linkedin": "https://linkedin.com/in/nunocoracao" },
+    { "twitter": "https://twitter.com/nunocoracao" },
+    { "instagram": "https://instagram.com/nunocoracao" },
+    { "medium": "https://medium.com/@nunocoracao" },
+    { "github": "https://github.com/nunocoracao" },
+    { "goodreads": "http://goodreads.com/nunocoracao" },
+    { "keybase": "https://keybase.io/nunocoracao" },
+    { "reddit": "https://reddit.com/user/nunoheart" }
+  ]
+}
```
Comment (quitterie-lcs) [inline data/authors/template.json:15]: Ditto

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+services:
+  hugo:
+    image: klakegg/hugo:0.105.0-ext-alpine
+    user: ${DOCKER_USER:-1000}
+    command: server -D
+    ports:
+      - "1313:1313"
+    volumes:
+      - .:/src
```
Comment (quitterie-lcs) [inline docker-compose.yml:11]: Ditto

File: static/site.webmanifest [hunk 10]
```diff
@@ -0,0 +1 @@
+{"name":"","short_name":"","icons":[{"src":"/android-chrome-192x192.png","sizes":"192x192","type":"image/png"},{"src":"/android-chrome-512x512.png","sizes":"512x512","type":"image/png"}],"theme_color":"#ffffff","background_color":"#ffffff","display":"standalone"}
```
Comment (quitterie-lcs) [inline static/site.webmanifest:1]: Ditto

**Review by quitterie-lcs — COMMENTED** [review]
Congrats! 
It was finally born :tada:

File: Makefile [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+.PHONY: logs
+
+status: ## an alias for "docker-compose ps"
+	@$(COMPOSE) ps
+.PHONY: status
+
+## -- Git
+git-modules-init: ## Init git modules
+	 git submodule update --init --recursive
+.PHONY: gitmodules-init
```
Comment (jmaupetit) [inline Makefile:45]: `init` or `update`? Should we create two targets instead?

File: Makefile [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+build: ## Build static blog
+	@$(COMPOSE_RUN_HUGO) -t $(THEME)
+.PHONY: build
+
+update-theme: ## Pull the theme submodules from its origin
+	cd ./themes/$(THEME)
+	git fetch origin
+	git pull origin --rebase --autostash
+.PHONY: update-theme
```
Comment (jmaupetit) [inline Makefile:70]: why not using `git submodule update` instead?

File: Makefile [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+# -- Articles
+new-post: ## Create a new post
+	bin/new-post
+.PHONY: new-post
```
Comment (jmaupetit) [inline Makefile:75]: Typing `new` dash `post` is not comfortable, what about the following?

```suggestion
new: ## Create a new post
	bin/new-post
.PHONY: new
```

File: README.md [hunk 14]
```diff
@@ -1,5 +1,76 @@
...
+
+## Authoring
+
+### Create a new post
+To create a new post, run the command :
+```sh
+make new-post
```
Comment (jmaupetit) [inline README.md:30]: Should be fixed if you apply my previous suggestion.

File: README.md [hunk 15]
```diff
@@ -1,5 +1,76 @@
...
+...
+authors:
+  - "<author_name>"
+---
+```
+
+### Further information
+
+Let's go further? Have a look at the documentation of [blowfish](https://nunocoracao.github.io/blowfish/docs/)
+and [Hugo](https://gohugo.io/documentation/).
```
Comment (jmaupetit) [inline README.md:76]: I would suggest to add a "Contributing" + "License" sections (see other projects).

**Review by jmaupetit — COMMENTED** [review]
:heart_eyes: ty :heart: 

A LICENSE file is missing for this project! Should we add a license for the content (CC-BY-SA) and another one for the code (MIT)?

Also I would suggest to add utilities to lint/format our markdown files using `prettier` so that the markdown syntax is homogeneous across posts.

:kissing_heart:

File: Makefile (hunk 13 above)
Comment (jbpenrath) [reply Makefile:75]: I suggest to use post instead. In this way, if we want later add commands to add author and so on, we will not have collisions.

File: Makefile (hunk 12 above)
Comment (jbpenrath) [reply Makefile:70]: Indeed I can use `git submodule update --remote`

File: Makefile (hunk 11 above)
Comment (jbpenrath) [reply Makefile:45]: Init as we use `--init` flag.
I replace the command `update-theme` by a command `git-modules-update`

**Comment by jbpenrath** [comment]
@jmaupetit I added our suggestions : 
- Two licences MIT for codebase and CC_BY_SA 4.0 for content
- Prettier to format post markdown files

File: Makefile (hunk 11 above)
Comment (jmaupetit) [reply Makefile]: Perfect.

**Pull request merged**